src/dipole/transition_switcher.rs

```diff
@@ -0,0 +1,63 @@
+extern crate rayon;
+extern crate specs;
+use crate::atom::{Atom, AtomicTransition, Kind};
+use crate::destructor::ToBeDestroyed;
+use crate::dipole::atom::AtomicDipoleTransition;
+use crate::laser::cooling::CoolingLight;
+use crate::laser::dipole_beam::DipoleLight;
+use specs::{Entities, Join, LazyUpdate, Read, ReadStorage, System};
+extern crate nalgebra;
+
+/// System that globally deletes all `CoolingLight` entities that do not have
+/// a `DipoleLight`. This could be used to speed up the simulation after a completed MOT power-ramp
+/// and transition into a pure dipole trap.
+pub struct DisableMOTBeamsSystem;
+
+impl<'a> System<'a> for DisableMOTBeamsSystem {
+    type SystemData = (
+        Entities<'a>,
+        ReadStorage<'a, CoolingLight>,
+        ReadStorage<'a, DipoleLight>,
+        Read<'a, LazyUpdate>,
+    );
+
+    fn run(&mut self, (ents, cooling, dipole, updater): Self::SystemData) {
+        for (entity, _cooling, _dipole) in (&ents, &cooling, !&dipole).join() {
+            updater.insert(entity, ToBeDestroyed);
+        }
+    }
+}
+
+/// Depending on the kind of `AtomicTransition`, this system associates the atom entities
+/// with an `AtomicDipoleTransition` so they are can be processed by the `ApplyDipoleForceSystem.`
+pub struct AttachAtomicDipoleTransitionToAtomsSystem;
```

Comment:
Is this the only place where `kind` is used?

If so, I _strongly_ advocate on behalf of the following changes:
(i) delete kind
(ii) delete this system

To explain my reasoning:
I think the purpose of this system is to make it so that you don't have to add an `AtomicDipoleTransition` to atoms when you create them. I understand in your case that could be annoying.

However, if I don't want to consider dipole traps, I shouldn't need to reserve that component storage! If I'm simulating only MOTs, I don't care about dipole components, and I should be able to create entities with only the components for MOTs - and that should be fine.

Secondly, although I don't have to add the `AtomicDipoleTransition` to my components, I now have to define a `Kind` on other components unrelated to this implementation. So the overhead for adding a new component has been removed, but now there is an API change and we require more definition when defining components used for laser cooling.

Third, I think this is not the right solution for the initial problem. The initial problem is that the API is becoming frustrating to use, because you have to define lots of components. If so, we should fix that by providing a high level API which takes care of these things for us - so I just say 'a rubidium atom here, please' and it does the rest, correctly sets up atoms, etc. See the high level API issue ticket.

Reply:
discussed that on teams - I agree